pool/pool.go

```diff
@@ -107,20 +109,56 @@ func (p *Pool) Update(b *blockchain.Block, bc *blockchain.BlockChain) bool {
 	return true
 }
 
-// PopTxns returns the the largest of l or size of pool transactions.
-// It selects the highest priority transactions, and removes them from the pool.
-func (p *Pool) PopTxns(l int) []*blockchain.Transaction {
-	if p.Len() == 0 {
-		return make([]*blockchain.Transaction, 0)
+// Pop returns the next transaction and removes it from the pool.
+func (p *Pool) Pop() *blockchain.Transaction {
+	if p.Len() > 0 {
+		next := p.GetN(0)
+		p.Delete(next)
+		return next
 	}
-	if p.Len() < l {
-		l = p.Len()
+	return nil
+}
+
+// Peek returns the next transaction and does not remove it from the pool.
+func (p *Pool) Peek() *blockchain.Transaction {
+	if p.Len() > 0 {
+		return p.GetN(0)
 	}
-	txns := make([]*blockchain.Transaction, l)
-	for i := 0; i < l; i++ {
-		t := p.GetN(i)
-		txns[i] = t
-		p.Delete(t)
+	return nil
+}
+
+// NextBlock produces a new block from the pool for mining.
+func (p *Pool) NextBlock(chain *blockchain.BlockChain,
```

Comment:
What do you think about calling it something like `MakeBlock` or `NewBlock`. I feel like `NextBlock